openssl/src/asn1.rs

```diff
@@ -95,6 +114,97 @@ foreign_type_and_impl_send_sync! {
     pub struct Asn1TimeRef;
 }
 
+impl Asn1TimeRef {
+    /// Find difference between two times
+    ///
+    /// This corresponds to [`ASN1_TIME_diff`].
+    ///
+    /// [`ASN1_TIME_diff`]: https://www.openssl.org/docs/man1.1.0/crypto/ASN1_TIME_diff.html
+    #[cfg(ossl102)]
+    pub fn diff(&self, compare: Option<&Self>) -> Result<TimeDiff, ErrorStack> {
+        let mut days = 0;
+        let mut seconds = 0;
+        let other = compare.map(Self::as_ptr).unwrap_or_else(ptr::null_mut);
+
+        let err = unsafe {
+            ffi::ASN1_TIME_diff(&mut days, &mut seconds, self.as_ptr(), other)
+        };
+
+        match err {
+            0 => Err(ErrorStack::get()),
+            _ => Ok(TimeDiff {
+                days: days,
+                secs: seconds,
+            }),
+        }
+    }
+
+    /// Compare two times
+    ///
+    /// This corresponds to [`ASN1_TIME_compare`] but is implemented using [`diff`] so that it is
+    /// also supported on older versions of OpenSSL.
+    ///
+    /// [`ASN1_TIME_compare`]: https://www.openssl.org/docs/man1.1.1/man3/ASN1_TIME_compare.html
+    /// [`diff`]: struct.Asn1TimeRef.html#method.diff
+    #[cfg(ossl102)]
+    pub fn compare(&self, other: &Self) -> Result<Ordering, ErrorStack> {
+        let d = self.diff(Some(other))?;
+        if d.days > 0 || d.secs > 0 {
+            return Ok(Ordering::Less);
+        }
+        if d.days < 0 || d.secs < 0 {
+            return Ok(Ordering::Greater);
+        }
+
+        Ok(Ordering::Equal)
+    }
+}
+
+#[cfg(ossl102)]
+impl Eq for Asn1TimeRef {}
```

Comment:
This isn't valid due to the error case, where you could in theory have a time that isn't equal to itself.

Reply:
True. Hadn't thought about that case. I will remove it.